### ISSUE TITLE
category visualisation error in Leaflet

### ISSUE THREAD

**Issue by Jeroen-GroeneBij** [qgis2web/qgis2web#821]
Hi,

using qgis 3.4.10 and q2w 3.8.0 I am having troubles correctly visualizing some layers.
Find attached the zip-file (map and style) to test. I have no idea what is going wrong.
The layer is a point layer, using category symbolization with a plain circle as symbol type.
Preview (and export) this as Leaflet shows all points in blue and with a different size.

The issue seems to be connected to the symbolization: changing it back to single symbol, the map is displayed correctly in preview.
Any idea what is going on?

[testhbombo.zip](https://github.com/tomchadwin/qgis2web/files/3783614/testhbombo.zip)

**Comment by Jeroen-GroeneBij** [comment]
Testing with the original point layer gave me the solution: the qgis styling file was made on an older version of this layer which used the field name Type (with a captital T). The new layer uses the field name type (with a lower-case t). Same data, I just changed to using lower-case consistently.
Loading that old style file into this new layer, QGIS apparently automatically changed the categorization to the lower case fieldname type , which is correct and works perfectly within QGIS, but QGIS gave no warning.
However, Q2W does not handle this the same way. I have to manually reconnect the categorization to the lower case fieldname type. Then Q2W is displaying the preview correctly again.

So QGIS is somehow 'helping' me by connecting the categorization to the most similar field, or doesn't make a difference between upper case and lower case fieldnames.
Q2W seems to be very accurate in upper case and lower case field names.

So, I have sorted this problem for myself. But I don't know if any changes have to be made to Q2W.

**Comment by tomchadwin** [comment]
Interesting - so you mean QGIS categorization is case-insensitive, but Leaflet's is case sensitive?

**Comment by Jeroen-GroeneBij** [comment]
At least with field names it seems to be, I'll run some test tomorrow. I don't know if this is true for the data it self. I will test this as well.

**Comment by Jeroen-GroeneBij** [comment]
Some interesting test results:

[testuplow.zip](https://github.com/tomchadwin/qgis2web/files/3787844/testuplow.zip)

I created a vectorlayer with a field name Type (upper case T). Then I created a style, using categorized symbology.
Next step is changing the field name to type (lower case T). This has no effect in QGIS, the categorized symbolization is applied correctly. Reloading layers, reloading styles, it is all consistent.
I even changed the field name to tyPe (upper case P), without effect.
Conclusion: QGIS is case-insenstive when considering fieldnames in applying a categorized style.

To support this conclusion, there is no way to see that the style applied actually doesn't match the field name. Qgis gives no warning.
However, taking this one step further, having two fields, type and Type is possible. But loading the style messes things up in QGIS: loading the style with the lower case type, QGIS applies the ELSE style to all records, like it is not sure which field to use, but not giving any warning of error. Loading the style with the upper case Type Qgis applies the style correctly and using the Type field.

Now for Q2W
When changing the fieldname I was expecting Q2W to come up with the default Leaflet blue for all records, because I presume it to be case-sensitive. and thus not recognizing the fieldname and thus no style would be exported. It didn't. In stead, Q2W showed all records in the style defined by the ELSE category.
So somehow the style is exported, but all records are considered to not match a category, just the ELSE category?
The same is true for changing the field name to tyPe.

To me, this behaviour of Q2W (Leaflet) is unexpected and illogical.


Considering the data, both QGIS and Q2W(Leaflet) are case-sensitive. All data in this test file are upper case, changing some to lower case will result in them not fitting the category and thus be part of the ELSE category. QGS and Q2W behave the same way.
